interp/interp_test.go

```diff
@@ -668,6 +668,22 @@ var runTests = []runTest{
 		"(fn() { :; }) & pwd >/dev/null",
 		"",
 	},
+	{
+		"x[0]=x; (echo ${x[0]}; x[0]=y; echo ${x[0]}); echo ${x[0]}",
+		"x\ny\nx\n",
+	},
+	{
+		`x["x"]=x; (x["x"]=y); echo ${x[0]}`,
+		"x\n",
+	},
+	{
+		"shopt -s expand_aliases; alias f='echo x'\nf\n(f\nalias f='echo y'\nf\n)\nf\n",
+		"x\nx\ny\nx\n",
+	},
+	{
+		"set -- a; echo $1; (echo $1; *[0]=b; echo $1); echo $1",
```

Comment:
err, that `*[0]` looks off. I don't think that should be valid syntax, and bash seems to agree with me. maybe a parser bug?

I don't think there is a way to modify a parameter directly. there is `set -- params...`, but that simply replaces the entire slice.

There's also `shift`, but that only keeps a subslice.

I don't have a problem with copying Params per se, but I don't think there's any case where it matters, so I'd go for not copying for the sake of avoiding work.

Reply:
Okay.  If `Params` is always replaced wholesale, and can't be modified in-place, then just copying it would be fine.

Reply:
FYI: https://github.com/mvdan/sh/commit/1e34cad3e3fdcb60d8a82aee5067395915153bc5